core/src/impl/Kokkos_Half_FloatingPointWrapper.hpp

```diff
@@ -215,10 +216,70 @@ cast_from_wrapper(const Kokkos::Experimental::bhalf_t& x);
 /************************** END forward declarations **************************/
 
 namespace Impl {
+
+template <typename FloatType>
+struct BitComparisonWrapper {
```

Comment:
would it make sense to add a staitic assert here to ensure the floatype is actually a float type?

Reply:
It's mostly templated so that they get different types for `half_t` and `bhalf_t`. All contexts where it's actually used use `bit_cast` which checks that the size of the types (this one and `[b]half_t::impl_type`) match. I could add a check in `FloatingPointWrapper` (which `half_t` and `bhalf_t` are aliases from) that the contained `impl_type` really has 16 bits but I don't think adding any checks here would be very useful.